lib/model/content.dart

```diff
@@ -658,27 +672,54 @@ class _ZulipContentParser {
       return UnimplementedBlockContentNode(htmlNode: divElement);
     }
 
-    final buffer = StringBuffer();
+    final spans = <CodeBlockSpanNode>[];
     for (int i = 0; i < mainElement.nodes.length; i++) {
+      final CodeBlockSpanNode span;
+
       final child = mainElement.nodes[i];
-      if (child is dom.Text) {
-        String text = child.text;
-        if (i == mainElement.nodes.length - 1) {
-          // The HTML tends to have a final newline here.  If included in the
-          // [Text] widget, that would make a trailing blank line.  So cut it out.
-          text = text.replaceFirst(RegExp(r'\n$'), '');
-        }
-        buffer.write(text);
-      } else if (child is dom.Element && child.localName == 'span') {
-        // TODO(#191) parse the code-highlighting spans, to style them
-        buffer.write(child.text);
-      } else {
-        return UnimplementedBlockContentNode(htmlNode: divElement);
+      switch (child) {
+        case dom.Text(:var text):
+          if (i == mainElement.nodes.length - 1) {
+            // The HTML tends to have a final newline here.  If included in the
+            // [Text] widget, that would make a trailing blank line.  So cut it out.
+            text = text.replaceFirst(RegExp(r'\n$'), '');
+          }
+          span = CodeBlockSpanNode(text: text, type: CodeBlockSpanType.text);
+
+        case dom.Element(:final text, :final classes, :final localName)
+          when localName == 'span'
+            // We assume that server will always provide us with a span element
+            // that has at most one class associated with the span element.
+            && classes.length <= 1:
```

Comment:
Skimming through [pygments html formatter](https://github.com/pygments/pygments/blob/ca2419cd7a7fdff73bb74a330cc49492420b1d25/pygments/formatters/html.py#L854-L858C35): It doesn't seem to emit a span element if `css_class` is empty. So, it's safe to say this will never happen; Removing that case.